src/renderer/api/endpoints/ingress.api.ts

```diff
@@ -117,6 +113,17 @@ export class Ingress extends KubeObject {
       ingress: ILoadBalancerIngress[];
     };
   };
+}
+
+export class Ingress extends KubeObject {
```

Comment:
What I meant is that this should be written as:

```typescript
export interface IngressSpec {
}

export interface IngressMetadata extends IKubeMetadata {
  tls: {
    secretName: string;
  }[];
  rules?: {
    host?: string;
    http: {
      paths: {
        path?: string;
        backend: IIngressBackend;
      }[];
    };
  }[];
  // extensions/v1beta1
  backend?: IExtensionsBackend;
  // networking.k8s.io/v1
  defaultBackend?: INetworkingBackend & {
    resource: {
      apiGroup: string;
      kind: string;
      name: string;
    }
  }
}

export interface IngressStatus {
  loadBalancer: {
    ingress: ILoadBalancerIngress[];
  };
}

export class Ingress extends KubeObject<IngressMetadata, IngressStatus, IngressSpec> {
   ...
}
```

Reply:
You can do that in follow-up PRs if you like.